pkg/resources/grant_privileges_to_database_role_identifier_test.go

```diff
@@ -0,0 +1,404 @@
+				Kind:             OnSchemaObjectDatabaseRoleGrantKind,
+				Data: &OnSchemaObjectGrantData{
+					Kind: OnAllSchemaObjectGrantKind,
+					OnAllOrFuture: &BulkOperationGrantData{
+						ObjectNamePlural: "TABLES",
+						Kind:             InSchemaBulkOperationGrantKind,
+						Schema:           sdk.Pointer(sdk.NewDatabaseObjectIdentifier("database-name", "schema-name")),
+					},
+				},
+			},
+		},
+		{
+			Name:       "grant database role on schema object with on future option",
+			Identifier: `"database-name"."database-role"|false|false|CREATE SCHEMA,USAGE,MONITOR|OnSchemaObject|OnFuture|TABLES`,
+			Expected: GrantPrivilegesToDatabaseRoleId{
+				DatabaseRoleName: sdk.NewDatabaseObjectIdentifier("database-name", "database-role"),
+				WithGrantOption:  false,
+				Privileges:       []string{"CREATE SCHEMA", "USAGE", "MONITOR"},
+				Kind:             OnSchemaObjectDatabaseRoleGrantKind,
+				Data: &OnSchemaObjectGrantData{
+					Kind: OnFutureSchemaObjectGrantKind,
+					OnAllOrFuture: &BulkOperationGrantData{
+						ObjectNamePlural: "TABLES",
+					},
+				},
+			},
+		},
+		{
+			Name:       "grant database role on schema object with on all option in database",
+			Identifier: `"database-name"."database-role"|false|false|CREATE SCHEMA,USAGE,MONITOR|OnSchemaObject|OnFuture|TABLES|InDatabase|"database-name-123"`,
+			Expected: GrantPrivilegesToDatabaseRoleId{
+				DatabaseRoleName: sdk.NewDatabaseObjectIdentifier("database-name", "database-role"),
+				WithGrantOption:  false,
+				Privileges:       []string{"CREATE SCHEMA", "USAGE", "MONITOR"},
+				Kind:             OnSchemaObjectDatabaseRoleGrantKind,
+				Data: &OnSchemaObjectGrantData{
+					Kind: OnFutureSchemaObjectGrantKind,
+					OnAllOrFuture: &BulkOperationGrantData{
+						ObjectNamePlural: "TABLES",
+						Kind:             InDatabaseBulkOperationGrantKind,
+						Database:         sdk.Pointer(sdk.NewAccountObjectIdentifier("database-name-123")),
+					},
+				},
+			},
+		},
+		{
+			Name:       "grant database role on schema object with on all option in schema",
+			Identifier: `"database-name"."database-role"|false|false|CREATE SCHEMA,USAGE,MONITOR|OnSchemaObject|OnFuture|TABLES|InSchema|"database-name"."schema-name"`,
+			Expected: GrantPrivilegesToDatabaseRoleId{
+				DatabaseRoleName: sdk.NewDatabaseObjectIdentifier("database-name", "database-role"),
+				WithGrantOption:  false,
+				Privileges:       []string{"CREATE SCHEMA", "USAGE", "MONITOR"},
+				Kind:             OnSchemaObjectDatabaseRoleGrantKind,
+				Data: &OnSchemaObjectGrantData{
+					Kind: OnFutureSchemaObjectGrantKind,
+					OnAllOrFuture: &BulkOperationGrantData{
+						ObjectNamePlural: "TABLES",
+						Kind:             InSchemaBulkOperationGrantKind,
+						Schema:           sdk.Pointer(sdk.NewDatabaseObjectIdentifier("database-name", "schema-name")),
+					},
+				},
+			},
+		},
+		{
+			Name:       "validation: grant database role not enough parts",
```

Comment:
Can you add tests for different parts empty? e.g.:
- empty privileges
- empty bools (e.g. with grant option)
- etc